app/internal/languages.py

```diff
@@ -0,0 +1,73 @@
+import gettext
+import os
+from pathlib import Path
+from typing import List
+
+from app import config
+from app.dependencies import templates
+
+LANGUAGE_DIR = "app/locales"
+LANGUAGE_DIR_TEST = "../app/locales"
+TRANSLATION_FILE = "base"
+
+
+def set_ui_language(language: str = None) -> None:
+    """Set the gettext translations to a given language.
+    If the language requested is not supported, the translations default
+    to the value of config.WEBSITE_LANGUAGE.
+
+    Args:
+        language (str, optional): a valid language code that follows RFC 1766.
+            Defaults to None.
+            See also the Language Code Identifier (LCID) Reference for a list of
+            valid language codes.
+
+    .. _RFC 1766:
+        https://tools.ietf.org/html/rfc1766.html
+
+    .. _Language Code Identifier (LCID) Reference:
+        https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-lcid/a9eac961-e77d-41a6-90a5-ce1a8b0cdb9c # noqa: E501
+    """
+
+    # TODO: Connect when user registration is completed.
+    # if not language:
+    #     language = _get_display_language(user_id: int)
+
+    if language not in _get_supported_languages():
+        language = config.WEBSITE_LANGUAGE
+
+    if Path(LANGUAGE_DIR).is_dir():
```

Comment:
I think we'd better monkey patch this one in our tests (it'll also increase your coverage rate)

Reply:
I'm not sure I did this correctly. I've added:
`languages.LANGUAGE_DIR = "../app/locales"`
to line 9 in conftest.py above from app.main import app, as anything after that crashes it (and this requires adding multiple noqa comments). But this seems much more ugly than the try-catch in languages.py. Is this how you wanted it to be?

Reply:
This is the only roadblock at the moment. I can't use monkeypatch in test_languages.py as by the time the code runs there, it already runs in conftest.py, so I need to keep the code as it is now, or add an ugly hack to the start of conftest. To me the current style is better, but fails codecov. Have any preference?

Reply:
I think I managed something, but if I'm really honest, I don't see the value of it, but I trust I'm missing something. Anywho, fixed!